indexing-service/src/main/java/io/druid/indexing/common/task/Task.java

```diff
@@ -178,7 +178,7 @@ default int getPriority()
    *
    * @throws Exception if this task failed
    */
-  TaskStatus run(TaskToolbox toolbox) throws Exception;
+  TaskStatusWithReports run(TaskToolbox toolbox) throws Exception;
```

Comment:
This might break the compatibility with the existing third-party task implementations.